cypress/integration/HW16/student_registration_form.spec.js

```diff
@@ -0,0 +1,126 @@
+        cy.get(locators.UPLOAD_PICTURE_LOCATOR).not('be.empty');
+
+        inputData.currentAddress(fakerVariables.RANDOM_CURRENT_ADDRESS)
+            .should('have.value', fakerVariables.RANDOM_CURRENT_ADDRESS);
+
+        inputData.state('Random');
+        cy.get(locators.SELECT_STATE_LOCATOR).not('be.empty');
+
+        inputData.city('Random');
+        cy.get(locators.SELECT_CITY_LOCATOR).not('be.empty');
+
+        buttons.submitButton();
+    });
+
+    it('Should have correct User Data in submit user data modal', () => {
+        check.checkSubmitForm('Student Name')
+            .should('include.text', `${fakerVariables.RANDOM_FIRST_NAME} ${fakerVariables.RANDOM_LAST_NAME}`);
+
+        check.checkSubmitForm('Student Email')
+            .should('have.text', fakerVariables.RANDOM_EMAIL);
+
+        check.checkSubmitForm('Gender')
+            .should('have.text', 'Male');
+
+        check.checkSubmitForm('Mobile')
+            .should('have.text', fakerVariables.RANDOM_PHONE_NUMBER);
+
+        check.checkSubmitForm('Date of Birth')
+            .should('include.text', `${fakerVariables.RANDOM_MONTH}`)
+            .should('include.text', `${fakerVariables.RANDOM_YEAR}`)
+            .should('include.text', `${fakerVariables.RANDOM_DAY_OF_THE_MONTH}`);
+
+        check.checkSubmitForm('Subjects')
+            .should('be.empty');
+
+        check.checkSubmitForm('Hobbies')
+            .should('have.text', 'Sports');
+
+        check.checkSubmitForm('Picture')
+            .should('have.text', 'photo.png');
+
+        check.checkSubmitForm('Address')
+            .should('have.text', fakerVariables.RANDOM_CURRENT_ADDRESS);
+
+        check.checkSubmitForm('State and City')
+            .not('be.empty');
+
+        buttons.closeButton();
+    });
+});
+
+describe('Filling out the "Student Registration Form" form with invalid data', () => {
+    beforeEach(() => {
+        cy.reload();
+    });
+
+    it('Check that the user can not submit the form with the invalid email', () => {
+        inputData.email('invalid Email');
+
+        buttons.submitButton();
+        cy.get(locators.USER_EMAIL_LOCATOR)
+            .should('have.css', 'border-color', cssVariables.ERROR_RED_COLOR)
+            .and('have.css', 'background-image');
+    });
+
+    it('Check that the user can not submit the form with the empty required fields', () => {
+        buttons.submitButton();
+
+        cy.get(locators.FIRST_NAME_LOCATOR)
+            .should('have.css', 'border-color', cssVariables.ERROR_RED_COLOR)
+            .and('have.css', 'background-image');
+
+        cy.get(locators.LAST_NAME_LOCATOR)
+            .should('have.css', 'border-color', cssVariables.ERROR_RED_COLOR)
+            .and('have.css', 'background-image');
+
+        cy.get(locators.USER_MOBILE_NUMBER_LOCATOR)
+            .should('have.css', 'border-color', cssVariables.ERROR_RED_COLOR)
+            .and('have.css', 'background-image');
+
```

Comment:
instead of dublicating code you can create only 1 function and then call it.
for example: 
let function checkSubmittingEmptyForm(fieldLocator) {
            cy.get(fieldLocator)
            .should('have.css', 'border-color', cssVariables.ERROR_RED_COLOR)
            .and('have.css', 'background-image');
}
And then just call it in your test passing there FIRST_NAME_LOCATOR etc. As a result you have only 3 lines of code instead of 9)

Reply:
Thanks for the feedback) I thought about this solution before, but in the end, I decided to write the checks themselves directly in the test file itself for ease of reading the code.